### DISCUSSION TITLE
Awaiting GPS Lock

### DISCUSSION THREAD

**Discussion by Reflexeye** [gopro/labs#866] [Q&A]
I'm using QR code to try and take Timelapse Photo (say every 30m) start 30 min before sunrise, stop 30m after sunset, and repeat forever. On scanning the QR the screen message is 'Awaiting GPS Lock'...then the camera powers off! Sadly, the timelapse photo sequence doesn't start. Any info gratefully received.
I've just discovered that the LATEST GoPro (12 Black) hasn't got GPS!...so I guess solar time is going to be impossible!?

**Comment by dnewman-gpsw** [comment]
You just need to set you location metadata manually on HERO12.  Manually is faster, anyway.

Use:  *LLTZ=latt,long,time_zone

For San Diego I use ```*LLTZ=32.707,-117.1576,-7```

**Comment by Reflexeye** [comment]
Ah - yes, I'll look deeper into the scripting! Thank you so much.

**Comment by Reflexeye** [comment]
I'm still getting gps lock message...
Anything obvious with mis-steps?

**Reply by dnewman-gpsw** [reply]
You haven't setup LLTZ correctly.  This works on HERO12 (just retested.)

![image](https://github.com/gopro/labs/assets/10856084/9e75b03b-9b5a-4111-ae54-04e983267cd1)

This temporarily set a location, and sleeps until sunset.  Remember you have to have a correct location for the sunset calculate to be correct-ish.

**Comment by Reflexeye** [comment]
...I've got rid of the message about gps lock by reboot. However, using
solar time is feeding back start times that don't correlate with settings.
E.G.:  '30 - Before - Sunset',  has created a start time of 14.09.53,
rather than 19.40.00 (approx).

On Tue, 16 Apr 2024 at 18:00, Jem McCluskey ***@***.***> wrote:

> I'm still getting gps lock message...
> Anything obvious with mis-steps?
>
> On Tue, 16 Apr 2024, 16:38 David Newman, ***@***.***> wrote:
>
>> You just need to set you location metadata manually on HERO12. Manually
>> is faster, anyway.
>>
>> Use: *LLTZ=latt,long,time_zone
>>
>> For San Diego I use *LLTZ=32.707,-117.1576,-7
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://github.com/gopro/labs/discussions/866#discussioncomment-9131853>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/BH2HEBS474EVKGJH4VBJVBLY5VAVVAVCNFSM6AAAAABGJHAUH2VHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4TCMZRHA2TG>
>> .
>> You are receiving this because you authored the thread.Message ID:
>> ***@***.***>
>>
>

-- 
Jem.

**Reply by dnewman-gpsw** [reply]
You have to have a correct LLTZ setting and correct camera date, time, TZ, for the solar calculator to work.  As this feature has excited for 4+ years, you can only assume this is a user error.

scan data/time https://gopro.github.io/labs/control/precisiontime/ 

use maps.google.com to get your location in latt and long.

**Comment by Reflexeye** [comment]
I've confirmed the correct times for camera and phone.
The cam screen reports the LLTZ feature (LLTZ=3).
I know my lat (51) and long (-3).
However, when lat, long flashes up on the cam screen after scanning the qr,
it reports lat and long the wrong way around...I'm guessing this is the
route of the problem I have, I suppose.

**Reply by dnewman-gpsw** [reply]
You must be adding spaces of something.  That fact is it displaying LLTZ=3 is proof of that.

If I use Taunton, UK with TZ BST (close to you)
```*LLTZ=51.0167856,-3.1401082,1```

My HERO12 says Sunset is at 20:07:44
https://www.timeanddate.com/astronomy/uk/taunton has it is at 20:10

This is within the expected precision for the simple calculator on camera.

**Comment by Reflexeye** [comment]
Many thanks - I'll have another look.

**Comment by Reflexeye** [comment]
The camera went flat overnight, as I forgot to turn it off. Strangely,
though - the lat/long/timezone is working with solar time and capturing
photos before sunset!...with interval of 5m I've put in for test purposes.

**Comment by Reflexeye** [comment]
I've now got this which I hope will work to take timelapse photos every 15
minutes starting before sunrise and ending after sunset and repeating every
day!
